frontend/src/features/admin-form/preview/PreviewFormProvider.tsx

```diff
@@ -0,0 +1,214 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { Helmet } from 'react-helmet-async'
+import { Text } from '@chakra-ui/react'
+import { differenceInMilliseconds, isPast } from 'date-fns'
+import { isEqual } from 'lodash'
+import get from 'lodash/get'
+import simplur from 'simplur'
+
+import { BasicField } from '~shared/types'
+import { FormAuthType, PreviewFormViewDto } from '~shared/types/form'
+
+import { usePreviewForm } from '~/features/admin-form/common/queries'
+import { FormNotFound } from '~/features/public-form/components/FormNotFound'
+import {
+  PublicFormContext,
+  SidebarSectionMeta,
+  SubmissionData,
+} from '~/features/public-form/PublicFormContext'
+import { useCommonFormProvider } from '~/features/public-form/PublicFormProvider'
+
+import { useTimeout } from '~hooks/useTimeout'
+import { HttpError } from '~services/ApiService'
+import Link from '~components/Link'
+
+import NotFoundErrorPage from '~pages/NotFoundError'
+import { trackVisitPublicForm } from '~features/analytics/AnalyticsService'
+import { useEnv } from '~features/env/queries'
+import {
+  RecaptchaClosedError,
+  useRecaptcha,
+} from '~features/recaptcha/useRecaptcha'
+import { FetchNewTransactionResponse } from '~features/verifiable-fields'
+
+interface PreviewFormProviderProps {
+  formId: string
+  children: React.ReactNode
+}
+
+export const PreviewFormProvider = ({
+  formId,
+  children,
+}: PreviewFormProviderProps): JSX.Element => {
+  // Once form has been submitted, submission data will be set here.
+  const [submissionData, setSubmissionData] = useState<SubmissionData>()
+  const [vfnTransaction, setVfnTransaction] =
+    useState<FetchNewTransactionResponse>()
+  const miniHeaderRef = useRef<HTMLDivElement>(null)
+  const { data, isLoading, error, ...rest } = usePreviewForm(
+    formId,
+    // Stop querying once submissionData is present.
+    /* enabled= */ !submissionData,
+  )
+  const { data: { captchaPublicKey } = {} } = useEnv(
+    /* enabled= */ !!data?.form.hasCaptcha,
+  )
+  const { hasLoaded, getCaptchaResponse, containerId } = useRecaptcha({
+    sitekey: data?.form.hasCaptcha ? captchaPublicKey : undefined,
+  })
```

Comment:
actually these may not need to be in the common hook, since we should disable captcha on preview hehe